cni/ipam/ipam_test.go

```diff
@@ -27,70 +30,191 @@ var ipamQueryResponse = "" +
 	"	</Interface>" +
 	"</Interfaces>"
 
-var localAsId string
-var poolId1 string
-var address1 string
+func TestIpam(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Ipam Suite")
+}
 
-// Wraps the test run with plugin setup and teardown.
-func TestMain(m *testing.M) {
-	var config common.PluginConfig
+// Handles queries from IPAM source.
+func handleIpamQuery(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(ipamQueryResponse))
+}
 
-	// Create a fake local agent to handle requests from IPAM plugin.
-	u, _ := url.Parse("tcp://" + ipamQueryUrl)
-	testAgent, err := common.NewListener(u)
+func parseResult(stdinData []byte) (*cniTypesCurr.Result, error) {
+	result := &cniTypesCurr.Result{}
+	err := json.Unmarshal(stdinData, result)
```

Comment:
for single returns, current style is to keep it on the same line

Reply:
Get, thanks!